### PR TITLE
[next] Fix edge cases with internal redirect sorting

### PR DESCRIPTION
**Pull request by ijjk**
This ensures we filter internal redirects more safely in versions that have the `internal` field in the `routes-manifest`. Our previous detection for trailing slash redirects could hit edge cases where it would match user redirects as well causing unexpected sorting/routing behavior. 

x-ref: NEXT-3840
x-ref: NDX-454

### CODE REVIEW

File: packages/next/test/integration/server-build/next.config.js [hunk 1]
```diff
@@ -3,4 +3,20 @@ module.exports = (phase, { defaultConfig }) => ({
   generateBuildId() {
     return 'testing-build-id';
   },
+
+  async redirects() {
+    // these routes are for testing trailing slash ordering
```
Comment (ztanner) [inline packages/next/test/integration/server-build/next.config.js:8]: is there a corresponding test assertion to go with this config?

Comment (ijjk) [reply packages/next/test/integration/server-build/next.config.js:8]: the Integration-1.test.js one

**Review by ztanner — APPROVED** [review]

File: packages/next/test/integration/server-build/next.config.js (hunk 1 above)
Comment (ztanner) [reply packages/next/test/integration/server-build/next.config.js:8]: my bad I somehow didn’t see that file on first look

**Comment by vikrantgupta25** [comment]
redirects for URLs without trailing slashes started failing because of this ? @ijjk

**Comment by ijjk** [comment]
@vikrantgupta25 Do you have trailingSlash: true configured?

**Comment by vikrantgupta25** [comment]
> Do you have trailingSlash: true configured?

yes we do have that as true, but the main issue being it used to work before and it stopped working post this release.
